lib/Gr4vyConfig.php

```diff
@@ -199,4 +212,215 @@ private static function getThumbprint($private_key) {
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_HTTPHEADER, array(
+                'Authorization: Bearer ' . $accessToken,
+                'Content-Type:application/json',
+                "X-GR4VY-MERCHANT-ACCOUNT-ID:" . $this->merchantAccountId
+            )
+        );
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "PUT");
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);
+        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, true);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        $responseData = curl_exec($ch);
+        if(curl_errno($ch)) {
+            return curl_error($ch);
+        }
+        curl_close($ch);
+        return json_decode($responseData, true);
+    }
+
+    private function delete($endpoint) {
+        $url = $this->host . $endpoint;
+        $scopes = array("*.read", "*.write");
+        $accessToken = self::getToken($this->privateKeyLocation, $scopes);
+
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_HTTPHEADER, array(
+                'Authorization: Bearer ' . $accessToken,
+                'Content-Type:application/json',
+                "X-GR4VY-MERCHANT-ACCOUNT-ID:" . $this->merchantAccountId
+            )
+        );
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "DELETE");
+        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, TRUE);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+        $responseData = curl_exec($ch);
+        $httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+        if(curl_errno($ch)) {
+            return curl_error($ch);
+        }
+        curl_close($ch);
+        if ($httpcode === 204) {
+            return array("success"=>true);    
+        }
+        return json_decode($responseData, true);
+    }
+    
+    public function addBuyer($buyer_request) {
+        $response = $this->post("/buyers", $buyer_request);
+        return $response;
+    }
+    public function getBuyer($buyer_id) {
+        $response = $this->get("/buyers/" . $buyer_id);
+        return $response;
+    }
+    public function updateBuyer($buyer_id, $buyer_request) {
+        $response = $this->put("/buyers/" . $buyer_id, $buyer_request);
+        return $response;
+    }
+    public function listBuyers() {
+        $response = $this->get("/buyers");
+        return $response;
+    }
+    public function deleteBuyer($buyer_id) {
+        $response = $this->delete("/buyers/" . $buyer_id);
+        return $response;
+    }
+
+    public function storePaymentMethod($payment_method_request) {
+        $response = $this->post("/payment-methods", $payment_method_request);
+        return $response;
+    }
+    public function getPaymentMethod($payment_method_id) {
+        $response = $this->get("/payment-methods/" . $payment_method_id);
+        return $response;
+    }
+    public function listPaymentMethods() {
+        $response = $this->get("/payment-methods");
+        return $response;
```

Comment:
@steve-gr4vy would we want to add support for query params?

Reply:
done.